src/odemis/driver/orsay.py

```diff
@@ -0,0 +1,1346 @@
+        else:
+            self.state._set_value(model.ST_RUNNING, force_write=True)
+
+    def _updatePosition(self, parameter=None, attributeName="Actual"):
+        """
+        parameter (Orsay Parameter): the parameter on the Orsay server to use to update the VA
+        attributeName (str): the name of the attribute of parameter which was changed
+
+        Reads the position of the GIS from the Orsay server and saves it in the position VA
+        """
+        if parameter is None:
+            parameter = self._positionPar
+        if parameter is not self._positionPar:
+            raise ValueError("Incorrect parameter passed to _updatePosition. Parameter should be "
+                             "datamodel.HybridGIS.PositionState. Parameter passed is %s." % parameter.Name)
+        if attributeName == "Actual":
+            pos = self._positionPar.Actual.lower()
+            logging.debug("Current position is %s." % pos)
+            self.position._set_value({"operational": pos == "work"}, force_write=True)
+        if self._positionPar.Actual == self._positionPar.Target:
+            logging.debug("Target position reached.")
+            self._operationalReached.set()
+        else:
+            self._operationalReached.clear()
+
+    def _updateGasOn(self, parameter=None, attributeName="Actual"):
+        """
+        parameter (Orsay Parameter): the parameter on the Orsay server to use to update the VA
+        attributeName (str): the name of the attribute of parameter which was changed
+
+        Reads the GIS gas flow state from the Orsay server and saves it in the gasOn VA
+        """
+        if parameter is None:
+            parameter = self._reservoirPar
+        if parameter is not self._reservoirPar:
+            raise ValueError("Incorrect parameter passed to _updateGasOn. Parameter should be "
+                             "datamodel.HybridGIS.ReservoirState. Parameter passed is %s." % parameter.Name)
+        if not attributeName == "Actual":
+            return
+        logging.debug("Gas flow is now %s." % self._reservoirPar.Actual)
+        new_value = self._reservoirPar.Actual.lower() == "open"
+        self.gasOn._value = new_value  # to not call the setter
+        self.gasOn.notify(new_value)
+
+    def _setGasOn(self, goal):
+        """
+        goal (bool): the goal state of the gas flow: (True: "open", False: "closed")
+        return (bool): the new state the gas flow: (True: "open", False: "closed")
+
+        Opens the GIS reservoir if argument goal is True. Closes it otherwise.
+        Also closes the reservoir if the position of the GIS is not operational.
+        """
+        if not self.position.value["operational"]:
+            logging.warning("Gas flow was attempted to be changed while not in working position.")
+            goal = False
+        if goal:
+            logging.debug("Starting gas flow.")
+            self._reservoirPar.Target = "OPEN"
```

Comment:
Please move all these strings (also for "operational") to constants. Eg: RESERVOIR_OPEN = "OPEN". This way, you can only mistake typing it once.